Recently added ReflectionFreeSerializationTest is very unstable

Recently added `ReflectionFreeSerializationTest` has been very unstable.

For instance see: https://github.com/quarkusio/quarkus/pull/44027 . But there were a lot of other failures lately.

Here is the output of the test:

```
2024-10-23T13:02:00.6037919Z 2024-10-23 13:02:00,591 ERROR [io.qua.res.rea.jac.run.map.NativeInvalidDefinitionExceptionMapper] (vert.x-eventloop-thread-1) com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class io.quarkus.mongodb.panache.reactive.ReactivePanacheMongoEntityBase and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: java.util.ArrayList[0])
2024-10-23T13:02:01.0187642Z [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 18.73 s <<< FAILURE! -- in io.quarkus.it.mongodb.panache.ReflectionFreeSerializationTest
2024-10-23T13:02:01.0192965Z [ERROR] io.quarkus.it.mongodb.panache.ReflectionFreeSerializationTest.testReactiveBookEntity -- Time elapsed: 1.060 s <<< ERROR!
2024-10-23T13:02:01.0194735Z java.lang.RuntimeException: 
2024-10-23T13:02:01.0196221Z com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
2024-10-23T13:02:01.0198080Z  at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1]
2024-10-23T13:02:01.0200161Z 	at io.restassured.internal.path.json.mapping.JsonPathJackson2ObjectDeserializer.deserialize(JsonPathJackson2ObjectDeserializer.java:30)
2024-10-23T13:02:01.0202739Z 	at io.restassured.path.json.mapping.JsonPathObjectDeserializer$deserialize.call(Unknown Source)
2024-10-23T13:02:01.0204786Z 	at io.restassured.internal.mapping.Jackson2Mapper.deserialize(Jackson2Mapper.groovy:58)
2024-10-23T13:02:01.0206059Z 	at io.restassured.mapper.ObjectMapper$deserialize.call(Unknown Source)
2024-10-23T13:02:01.0207526Z 	at io.restassured.internal.mapping.ObjectMapping.parseWithJackson2(ObjectMapping.groovy:254)
2024-10-23T13:02:01.0208976Z 	at io.restassured.internal.mapping.ObjectMapping$parseWithJackson2$0.callStatic(Unknown Source)
2024-10-23T13:02:01.0210334Z 	at io.restassured.internal.mapping.ObjectMapping.deserialize(ObjectMapping.groovy:56)
2024-10-23T13:02:01.0211587Z 	at io.restassured.internal.mapping.ObjectMapping$deserialize.call(Unknown Source)
2024-10-23T13:02:01.0213126Z 	at io.restassured.internal.RestAssuredResponseOptionsGroovyImpl.as(RestAssuredResponseOptionsGroovyImpl.groovy:229)
2024-10-23T13:02:01.0214803Z 	at io.restassured.internal.RestAssuredResponseOptionsGroovyImpl$as$2.callCurrent(Unknown Source)
2024-10-23T13:02:01.0216630Z 	at io.restassured.internal.RestAssuredResponseOptionsGroovyImpl.as(RestAssuredResponseOptionsGroovyImpl.groovy:246)
2024-10-23T13:02:01.0218379Z 	at io.restassured.internal.RestAssuredResponseOptionsImpl.as(RestAssuredResponseOptionsImpl.java:186)
2024-10-23T13:02:01.0220449Z 	at io.quarkus.it.mongodb.panache.reactive.ReactiveMongodbPanacheResourceTest.callReactiveBookEndpoint(ReactiveMongodbPanacheResourceTest.java:107)
2024-10-23T13:02:01.0222682Z 	at io.quarkus.it.mongodb.panache.ReflectionFreeSerializationTest.testReactiveBookEntity(ReflectionFreeSerializationTest.java:29)
2024-10-23T13:02:01.0224513Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
2024-10-23T13:02:01.0225990Z 	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:971)
2024-10-23T13:02:01.0227812Z 	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:821)
2024-10-23T13:02:01.0229276Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2024-10-23T13:02:01.0230315Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2024-10-23T13:02:01.0231858Z Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
2024-10-23T13:02:01.0233349Z  at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1]
2024-10-23T13:02:01.0234811Z 	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
2024-10-23T13:02:01.0236339Z 	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:5008)
2024-10-23T13:02:01.0237972Z 	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4910)
2024-10-23T13:02:01.0239293Z 	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3860)
2024-10-23T13:02:01.0241153Z 	at io.restassured.internal.path.json.mapping.JsonPathJackson2ObjectDeserializer.deserialize(JsonPathJackson2ObjectDeserializer.java:28)
2024-10-23T13:02:01.0242617Z 	... 18 more
```

--- a/extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/JacksonCodeGenerator.java
+++ b/extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/JacksonCodeGenerator.java
@@ -70,7 +70,7 @@ public abstract class JacksonCodeGenerator {
 
     private Optional<String> create(ClassInfo classInfo) {
         String beanClassName = classInfo.name().toString();
-        if (vetoedClassName(beanClassName) || !generatedClassNames.add(beanClassName)) {
+        if (vetoedClass(classInfo, beanClassName) || !generatedClassNames.add(beanClassName)) {
             return Optional.empty();
         }
 
@@ -145,7 +145,11 @@ public abstract class JacksonCodeGenerator {
         return name.substring(0, 1).toUpperCase() + name.substring(1);
     }
 
-    protected static boolean vetoedClassName(String className) {
+    protected static boolean vetoedClass(ClassInfo classInfo, String className) {
+        return classInfo.isAbstract() || classInfo.isInterface() || vetoedClassName(className);
+    }
+
+    private static boolean vetoedClassName(String className) {
         return className.startsWith("java.") || className.startsWith("jakarta.") || className.startsWith("io.vertx.core.json.");
     }
 
@@ -203,11 +207,9 @@ public abstract class JacksonCodeGenerator {
     }
 
     private void registerTypeToBeGenerated(String typeName) {
-        if (!vetoedClassName(typeName)) {
-            ClassInfo classInfo = jandexIndex.getClassByName(typeName);
-            if (classInfo != null && shouldGenerateCodeFor(classInfo)) {
-                toBeGenerated.add(classInfo);
-            }
+        ClassInfo classInfo = jandexIndex.getClassByName(typeName);
+        if (classInfo != null && !vetoedClass(classInfo, typeName) && shouldGenerateCodeFor(classInfo)) {
+            toBeGenerated.add(classInfo);
         }
     }
 
@@ -312,8 +314,14 @@ public abstract class JacksonCodeGenerator {
         }
 
         boolean hasUnknownAnnotation() {
-            return annotations.keySet().stream()
-                    .anyMatch(ann -> ann.startsWith("com.fasterxml.jackson.") && !ann.equals(JsonProperty.class.getName()));
+            return annotations.keySet().stream().anyMatch(FieldSpecs::isUnknownAnnotation);
+        }
+
+        private static boolean isUnknownAnnotation(String ann) {
+            if (ann.startsWith("com.fasterxml.jackson.")) {
+                return !ann.equals(JsonProperty.class.getName());
+            }
+            return false;
         }
 
         ResultHandle toValueWriterHandle(BytecodeCreator bytecode, ResultHandle valueHandle) {

